components/org.wso2.carbon.identity.mgt.endpoint/src/main/java/org/wso2/carbon/identity/mgt/endpoint/IdentityManagementServiceUtil.java

```diff
@@ -31,6 +31,7 @@
 import org.wso2.carbon.utils.multitenancy.MultitenantUtils;
 import org.wso2.securevault.SecretResolver;
 import org.wso2.securevault.SecretResolverFactory;
+import org.wso2.securevault.commons.MiscellaneousUtil;
```

Comment:
What is MiscellaneousUtil, can we do any damn thing ?
Ideally any util should have a purpose, and named accordingly.

Reply:
There is a MiscellaneousUtil class[1] in carbon-secvault. This class is a combination of different resolve methods for decrypting secrets(Eg: XML elements, texts and OMAttribute)

[1] - https://github.com/wso2/carbon-secvault/blob/1.x/src/main/java/org/wso2/securevault/commons/MiscellaneousUtil.java

Reply:
Ack